spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/converter/UniProtEntryCommentsConverter.java

```diff
@@ -638,113 +638,189 @@ private Set<String> fetchEvidences(Comment comment) {
+        }
+        return evidences;
+    }
+
+    private Set<String> fetchCatalyticActivity(CatalyticActivityComment caComment) {
+        Set<String> evidences = new HashSet<>();
+        if (caComment.hasReaction()) {
+            evidences.addAll(
+                    UniProtEntryConverterUtil.extractEvidence(
+                            caComment.getReaction().getEvidences()));
+        }
+        if (caComment.hasPhysiologicalReactions()) {
+            List<Evidence> physiologicalEvidences =
+                    caComment.getPhysiologicalReactions().stream()
+                            .filter(PhysiologicalReaction::hasEvidences)
+                            .flatMap(reaction -> reaction.getEvidences().stream())
+                            .collect(Collectors.toList());
+            evidences.addAll(UniProtEntryConverterUtil.extractEvidence(physiologicalEvidences));
+        }
+        return evidences;
+    }
+
+    private Set<String> fetchCofactorEvidences(CofactorComment cofactorComment) {
+        Set<String> evidences = new HashSet<>();
+        if (cofactorComment.hasCofactors()) {
+            cofactorComment.getCofactors().stream()
+                    .flatMap(
+                            cofactor ->
+                                    UniProtEntryConverterUtil.extractEvidence(
+                                            cofactor.getEvidences())
+                                            .stream())
+                    .forEach(evidences::add);
+        }
+
+        if ((cofactorComment.hasNote()) && (cofactorComment.getNote().hasTexts())) {
+            evidences.addAll(getTextsEvidence(cofactorComment.getNote().getTexts()));
+        }
+        return evidences;
+    }
+
+    private Set<String> fetchRNAEditingEvidences(RnaEditingComment reComment) {
+        Set<String> evidences = new HashSet<>();
+        if (reComment.hasPositions()) {
+            evidences.addAll(
+                    UniProtEntryConverterUtil.extractEvidence(
+                            reComment.getPositions().stream()
+                                    .flatMap(val -> val.getEvidences().stream())
+                                    .collect(Collectors.toList())));
+        }
+        if (reComment.hasNote() && reComment.getNote().hasTexts()) {
+            evidences.addAll(getTextsEvidence(reComment.getNote().getTexts()));
+        }
+        return evidences;
+    }
+
+    private Set<String> fetchDiseaseEvidences(DiseaseComment diseaseComment) {
+        Set<String> evidences = new HashSet<>();
+        if (diseaseComment.hasDefinedDisease()) {
+            evidences.addAll(
+                    UniProtEntryConverterUtil.extractEvidence(
+                            diseaseComment.getDisease().getEvidences()));
+            if (diseaseComment.hasNote() && diseaseComment.getNote().hasTexts()) {
+                evidences.addAll(getTextsEvidence(diseaseComment.getNote().getTexts()));
+            }
         }
```

Comment:
Every time i see this class.. I am like when are we going to refactor this.

Reply:
I created a Jira for it: https://www.ebi.ac.uk/panda/jira/browse/TRM-29257